src/ast/ast.h

```diff
@@ -1098,31 +1074,22 @@ class For : public Node {
 public:
   explicit For(ASTContext &ctx,
                Variable *decl,
-               Map *map,
-               StatementList &&stmts,
-               Location &&loc)
-      : Node(ctx, std::move(loc)),
-        decl(decl),
-        iterable(map),
-        stmts(std::move(stmts)) {};
-  explicit For(ASTContext &ctx,
-               Variable *decl,
-               Range *range,
-               StatementList &&stmts,
+               Iterable iterable,
+               BlockExpr *block,
                Location &&loc)
       : Node(ctx, std::move(loc)),
         decl(decl),
-        iterable(range),
-        stmts(std::move(stmts)) {};
+        iterable(iterable),
+        block(block) {};
```

Comment:
Why did this need to change?

Reply:
The `bare_block` structure in the parser was needed to avoid a reduce/reduce conflict. The `block_expr` variant of that is already a `BlockExpr*` so it is simplest to have the `bare_block` be a `BlockExpr*`.

I think it's simpler if this is just a block rather than having a hard coded `StatementList` anyways, but it makes the construction from the parser slightly cleaner.

Reply:
> it makes the construction from the parser slightly cleaner

That's true. The reading of it is a little odd though as these things don't produce expressions. Non-blocking, just thinking it through.